File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> li = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        if (root == null)
        {
            return li;
        }
        q.add(root);
        while (q.size() > 0)
        {
            int size = q.size();
            List<Integer> lii = new ArrayList<Integer>();
            for (int i = 0;i<size;i++)
            {
                TreeNode node = q.poll();
                lii.add(node.val);
                if (node.left != null)
                {
                    q.add(node.left);
                }
                if (node.right != null)
                {
                    q.add(node.right);
                }
            }
            li.add(lii);
        }
        return li;
    }
}

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[List[int]]
        """
        if root == None:
            return []
        queue = deque()
        queue.append(root)
        res = []
        while queue:
            size = len(queue)
            lesres = []
            while size > 0:
                size -= 1
                temp = queue.popleft()
                lesres.append(temp.val)
                if temp.left: queue.append(temp.left)
                if temp.right: queue.append(temp.right)
            res.append(lesres)
        return res